ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/metrics/DeltaFilesMetricReporter.java

```diff
@@ -212,11 +215,41 @@ public static void mergeDeltaFilesStats(AcidDirectory dir, long checkThresholdIn
         }
       }
     }
+
+    logDeltaDirMetrics(dir, conf, numObsoleteDeltas, numDeltas, numSmallDeltas);
+
     String path = getRelPath(dir);
     newDeltaFilesStats(numObsoleteDeltas, numDeltas, numSmallDeltas)
       .forEach((type, cnt) -> deltaFilesStats.computeIfAbsent(type, v -> new HashMap<>()).put(path, cnt));
   }
 
+  private static void logDeltaDirMetrics(AcidDirectory dir, Configuration conf, int numObsoleteDeltas, int numDeltas,
+      int numSmallDeltas) {
+    long loggerFrequency = HiveConf
+        .getTimeVar(conf, HiveConf.ConfVars.HIVE_COMPACTOR_ACID_METRICS_LOGGER_FREQUENCY, TimeUnit.MILLISECONDS);
+    if (loggerFrequency <= 0) {
+      return;
+    }
+    long currentTime = System.currentTimeMillis();
+    if (lastSuccessfulLoggingTime == 0 || currentTime >= lastSuccessfulLoggingTime + loggerFrequency) {
```

Comment:
I think the first part should be: loggerFrequency == 0

Reply:
Could you please elaborate on this? If `loggerFrequence = 0` the method returns. 
`lastSuccessfullLoggingTime == 0` means that the `logDeltaDirMetrics` is called for the very first time.

Reply:
got it now:)